modules/activitypub/iri.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package activitypub
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	repo_model "code.gitea.io/gitea/models/repo"
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/setting"
+
+	ap "github.com/go-ap/activitypub"
+)
+
+// Returns the username corresponding to a Person actor IRI
+func personIRIToName(personIRI ap.IRI) (string, error) {
+	personIRISplit := strings.Split(personIRI.String(), "/")
+	if len(personIRISplit) < 3 {
+		return "", errors.New("Not a Person actor IRI")
```

Comment:
> Confusing error, "Invalid IRI"?

It could still be a valid IRI though. This check is to guarantee the IRI at least looks like `https://instance.com/user`, so there's a protocol, instance, and username in the IRI to do "reverse WebFinger".

Reply:
this does handle the `//` correctly, right?

Reply:
@zseri Which `//`?

Reply:
`personIRI.String()` returns a string, and @Ta180m's comment indicates that the string should have the format `SCHEME://INSTANCE.DOMAIN/USER`, which obviously contains a `//` after `SCHEME:`; does go have a proper URI parser though, that might be more appropriate...

Reply:
Ah yeah, so that's correctly. See line 26 and 27, so just to be safe:

Example output of `String()`:
``https://git.gusted.xyz/gitea/Gusted`
In this case I'm running a Gitea sever on the `git.gusted.xyz` subdomain and with the `/gitea` path.

So using
```go
strings.Split(personIRI.String(), "/") // returns []string{"https:", "", "git.gusted.xyz", "gitea", "Gusted"}
```

So:
```go
instance := "git.gusted.xyz"
name := "Gusted"
```

@Ta180m Now I'm not quite sure here, but it seems like we might need to take into account the `/gitea` path as part of the instance.

Reply:
but that also means that `< 3` is incorrect and should be `< 4`, right?

Reply:
Yup pretty much the smallest accept length is

```go
[]string{"https:", "", "git.gusted.xyz", "Gusted"} // length 4. 
```